extensions/sns/src/utils.rs

```diff
@@ -1,18 +1,38 @@
 //! Utility functions for the SNS extension.
-
-use anyhow::{anyhow, Result};
+use anyhow::{Context, Result};
+use dfx_core::interface::builder::IdentityPicker;
+use dfx_core::DfxInterface;
 use ic_agent::agent::Agent;
 
-/// Gets the agent corresponding to the given IC URL.
-fn get_agent(ic_url: &str) -> Result<Agent> {
-    Agent::builder()
-        .with_url(ic_url)
-        .build()
-        .map_err(|e| anyhow!(e))
+/// Gets an agent for a given network and identity. This is similar to the code DFX uses internally to get an agent.
+/// If no identity is provided, it will use the identity currently selected in the DFX CLI.
+pub async fn get_agent(network_name: &str, identity: Option<String>) -> Result<Agent> {
+    let interface = dfx_interface(network_name, identity)
+        .await
+        .context("Failed to get dfx interface")?;
+    Ok(interface.agent().clone())
 }
 
-/// Gets the agent for IC mainnet.
-pub fn get_mainnet_agent() -> Result<Agent> {
-    let ic_url = "https://ic0.app/";
-    get_agent(ic_url)
+/// Gets a dfx interface for a given network and identity. This is similar to the code DFX uses internally to build the interface.
+/// So this function allows the DFX SNS Extension to use the same interface as DFX itself.
+/// If no identity is provided, it will use the identity currently selected in the DFX CLI.
+pub async fn dfx_interface(network_name: &str, identity: Option<String>) -> Result<DfxInterface> {
```

Comment:
I remember seeing this code already somewhere else - is it possible to re-use that as a dependency, or will those change for unrelated reasons?

Reply:
We are planning on having it change for unrelated reasons. Down the line we want the monorepo CLI to have slightly different defaults than the DFX SNS extension